Makefile

```diff
@@ -35,37 +38,35 @@ else
 checkprefix: ;
 endif
 
-BUILD_TYPE ?= $(shell (type ninja > /dev/null 2>&1 && echo "Ninja") || \
+CMAKE_GENERATOR ?= $(shell (command -v ninja > /dev/null 2>&1 && echo "Ninja") || \
     echo "Unix Makefiles")
 DEPS_BUILD_DIR ?= .deps
 ifneq (1,$(words [$(DEPS_BUILD_DIR)]))
   $(error DEPS_BUILD_DIR must not contain whitespace)
 endif
 
 ifeq (,$(BUILD_TOOL))
-  ifeq (Ninja,$(BUILD_TYPE))
+  ifeq (Ninja,$(CMAKE_GENERATOR))
     ifneq ($(shell $(CMAKE_PRG) --help 2>/dev/null | grep Ninja),)
       BUILD_TOOL := ninja
     else
       # User's version of CMake doesn't support Ninja
       BUILD_TOOL = $(MAKE)
-      BUILD_TYPE := Unix Makefiles
```

Comment:
why remove this? the condition on line 50 isn't checked anywhere else. If CMAKE_GENERATOR is "ninja" but that condition fails, then we have a regression

Reply:
It's because it's done in a slightly convoluted logic: https://github.com/neovim/neovim/blob/1090ac568272565879b2f0d2c61c94d4884e34f7/Makefile#L41-L42 
I had figured that you wouldn't actually have `ninja` on the system while simultaneously have an old/(ancient?) `cmake`.
But you're right, there's an actual chance of a regression here, although unlikely. 

If you want me to change it, then my vote is to start the change from the original line that I linked, and actually only change the default value of the variable, "Unix Makefile", when both requirements are met 1) compatible cmake, 2) ninja is available

Reply:
> If you want me to change it, then my vote is to start the change from the original line that I linked, and actually only change the default value of the variable, "Unix Makefile", when both requirements are met 1) compatible cmake, 2) ninja is available

it's not worth the time to think about it and potentially regress, until we have an actual need to change the logic.